### DISCUSSION TITLE
Log Retention - Automatic Deletion

### DISCUSSION THREAD

**Discussion by veraxcyber** [Security-Onion-Solutions/securityonion#4540] [Unsupported Versions]
Hello all! Hope you are well! 

What is the best way to implement a Log Retention period with Security Onion please? With the addition of preventing the hard drive from reach max capacity.

Many Thanks!

Andrew

**Comment by facyber** [comment]
Do you mean retention of log files or data itself?

For log files there are already predefined log rotations.

For disk size getting full, by default Stenographer is deleting old PCAP files once disk is 90% full. You can find more about Stenographer [here](https://docs.securityonion.net/en/2.3/stenographer.html#stenographer).

For Elasticsearch, there is also defined value how much GB of ES data should be kept before cleaning process start. For ES limit check [this](https://docs.securityonion.net/en/2.3/curator.html?highlight=log_size_limit#configuration) page.

You will have to find a balance between your PCAP and ES data and optimize it according to your environment.

**Comment by veraxcyber** [comment]
Hey!

Really appreciate the response and you have answered my questions, thanks for this!
